src/numeric_index.c

```diff
@@ -156,26 +166,21 @@ NRN_AddRv NumericRangeNode_Add(NumericRangeNode *n, t_docId docId, double value)
       // if there was a split it means our max depth has increased.
       // we are too deep - we don't retain this node's range anymore.
       // this keeps memory footprint in check
-      if (++n->maxDepth > NR_MAX_DEPTH && n->range) {
-        rv.sz -= n->range->invertedIndexSize;
-        rv.numRecords -= n->range->entries->numDocs;
-        InvertedIndex_Free(n->range->entries);
-        array_free(n->range->values);
-        rm_free(n->range);
-        n->range = NULL;
+      if (++n->maxDepth > NR_MAX_DEPTH_RANGE && n->range) {
```

Comment:
I thought we said not to change it if the new config val is on

Reply:
@MeirShpilraien - done